BlueprintUI/Sources/Layout/ConstrainedSizeFraction.swift

```diff
@@ -0,0 +1,94 @@
+import UIKit
+
+/// Constrains the size of the content element to a fraction of the parent's layout space.
+public struct ConstrainedSizeFraction: Element {
+    /// The element being constrained.
+    public var wrappedElement: Element
+    /// The fraction of the parent's horizontal layout space the content element should occupy. `nil` if the width is
+    /// unconstrained.
+    public var fractionalWidth: CGFloat?
+    /// The fraction of the parent's vertical layout space the content element should occupy. `nil` if the height is
+    /// unconstrained.
+    public var fractionalHeight: CGFloat?
+
+    /// Initializes with the given properties.
+    ///
+    /// - parameters:
+    ///   - fractionalWidth: The fraction of the parent's horizontal layout space the content element should occupy. `nil` if
+    ///   the width is unconstrained. By default, `nil`.
+    ///   - fractionalHeight: The fraction of the parent's vertical layout space the content element should occupy. `nil` if
+    ///   the height is unconstrained. By default, `nil`.
+    ///   - wrapping: The content element.
+    public init(fractionalWidth: CGFloat? = nil, fractionalHeight: CGFloat? = nil, wrapping wrappedElement: Element) {
+        precondition(
+            fractionalWidth.map { $0 >= 0 && $0 <= 1} ?? true,
+            "The provided width fraction must be a value in the range of `0...1`."
+        )
+        precondition(
+            fractionalHeight.map { $0 >= 0 && $0 <= 1} ?? true,
+            "The provided height fraction must be a value in the range of `0...1`."
+        )
+
+        self.fractionalWidth = fractionalWidth
+        self.fractionalHeight = fractionalHeight
+        self.wrappedElement = wrappedElement
+    }
+
+    public var content: ElementContent {
+        ElementContent(
+            child: wrappedElement,
+            layout: Layout(width: fractionalWidth, height: fractionalHeight)
+        )
+    }
+
+    public func backingViewDescription(bounds: CGRect, subtreeExtent: CGRect?) -> ViewDescription? {
+        return nil
+    }
+
+    private struct Layout: SingleChildLayout {
+        var width: CGFloat?
+        var height: CGFloat?
+
+        func measure(in sizeConstraint: SizeConstraint, child: Measurable) -> CGSize {
+            /// Get the target layout space, constraining it by the specified fractions.
+            let targetWidth = width.map { $0 * sizeConstraint.width.maximum }
```

Comment:
My approach here is exactly the same as `ConstrainedSize's`.

```
let maximumConstraint = SizeConstraint(
    width: .init(self.width.applied(to: constraint.width.maximum)),
    height: .init(self.height.applied(to: constraint.height.maximum))
)
```

Reply:
Hmm, I think `ConstrainedSize` is probably wrong too now that you point it out, but it hasn't caused any problems in practice because it only affects `atLeast` constraints?

Regardless, I think it is a bit nonsensical to do fraction math on `unconstrained`. 50% of an infinite canvas is still an infinite canvas.